### PR TITLE
Cmake dep refactor

### PR DESCRIPTION
**Pull request by petersteneteg**

### CODE REVIEW

File: cmake/installutils.cmake [hunk 1]
```diff
@@ -45,72 +50,161 @@ if(APPLE)
...
-            COMPONENT ${comp})
+    ivw_default_install_targets(${ARGN})
+endfunction()
+
+define_property(
+    GLOBAL PROPERTY IVW_INSTALL_LIST INHERITED
+    BRIEF_DOCS "List of intstallation components to install for this target"
+    FULL_DOCS "List of intstallation components to install for this target"
+)
+define_property(
```
Comment (ResearchDaniel) [inline cmake/installutils.cmake:83]: 2 Duplicates

File: cmake/installutils.cmake [hunk 2]
```diff
@@ -45,72 +50,161 @@ if(APPLE)
...
 # Add folder to module pack
 macro(ivw_add_to_module_pack folder)
-    get_filename_component(FOLDER_NAME ${CMAKE_CURRENT_SOURCE_DIR} NAME)
+    get_filename_component(module ${CMAKE_CURRENT_SOURCE_DIR} NAME)
     install(
         DIRECTORY ${folder}
-        DESTINATION ${IVW_RESOURCE_INSTALL_PREFIX}modules/${FOLDER_NAME}
-        COMPONENT ${_cpackName}
+        DESTINATION ${IVW_RESOURCE_INSTALL_PREFIX}modules/${module}
+        COMPONENT Application
```
Comment (ResearchDaniel) [inline cmake/installutils.cmake]: What about development?

File: cmake/installutils.cmake [hunk 3]
```diff
@@ -45,72 +50,161 @@ if(APPLE)
...
+    list(APPEND install_list 
+        "${CMAKE_CURRENT_BINARY_DIR}|%|${CMAKE_PROJECT_NAME}|%|Application|%|/"
+        "${CMAKE_CURRENT_BINARY_DIR}|%|${CMAKE_PROJECT_NAME}|%|Datasets|%|/"
+        "${CMAKE_CURRENT_BINARY_DIR}|%|${CMAKE_PROJECT_NAME}|%|Testing|%|/"
+        "${CMAKE_CURRENT_BINARY_DIR}|%|${CMAKE_PROJECT_NAME}|%|Development|%|/"
+    )
+    set_property(${scope} PROPERTY IVW_INSTALL_LIST ${install_list})
 endfunction()
 
+function(ivw_filter_install_list)
```
Comment (ResearchDaniel) [inline cmake/installutils.cmake:130]: Documentation would be good, not obvious how to use it.

File: cmake/installutils.cmake [hunk 4]
```diff
@@ -45,72 +50,161 @@ if(APPLE)
...
+    BRIEF_DOCS "List of intstallation components to install for this target"
+    FULL_DOCS "List of intstallation components to install for this target"
+)
+define_property(
+    TARGET PROPERTY IVW_INSTALL_LIST INHERITED
+    BRIEF_DOCS "List of intstallation components to install for this target"
+    FULL_DOCS "List of intstallation components to install for this target"
+)
+
+function(ivw_append_install_list)
```
Comment (ResearchDaniel) [inline cmake/installutils.cmake:95]: Documentation would be good, not obvious how to use it.

File: cmake/installutils.cmake [hunk 5]
```diff
@@ -45,72 +50,161 @@ if(APPLE)
...
+    install(
+        DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/data
+        DESTINATION ${IVW_RESOURCE_INSTALL_PREFIX}modules/${module}/data
+        COMPONENT Datasets
+        OPTIONAL
+    )
+    install(
+        DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/docs
+        DESTINATION ${IVW_RESOURCE_INSTALL_PREFIX}modules/${module}/docs
+        COMPONENT Application
```
Comment (ResearchDaniel) [inline cmake/installutils.cmake:201]: Development?

File: cmake/installutils.cmake [hunk 6]
```diff
@@ -45,72 +50,161 @@ if(APPLE)
...
+
 #--------------------------------------------------------------------
 # A helper funtion to install module targets
 function(ivw_default_install_targets)
-    ivw_default_install_comp_targets(modules ${ARGN})
+    foreach(target IN LISTS ARGN)
+        install(TARGETS ${target} EXPORT "${target}"
+            RUNTIME        # DLLs, Exes, 
+                COMPONENT Application 
+                DESTINATION ${IVW_RUNTIME_INSTALL_DIR}
```
Comment (ResearchDaniel) [inline cmake/installutils.cmake:164]: Can’t we use the cmake variable for this?

File: cmake/templates/inviwocommondefines_template.h [hunk 7]
```diff
@@ -1,11 +1,10 @@
...
-
-#define IVW_VERSION std::string("@IVW_VERSION@")
-
-#define IVW_MAJOR_VERSION @IVW_MAJOR_VERSION@
-#define IVW_MINOR_VERSION @IVW_MINOR_VERSION@
-#define IVW_PATCH_VERSION @IVW_PATCH_VERSION@
-#define IVW_BUILD_VERSION @IVW_BUILD_VERSION@
+namespace inviwo::build {
+	inline constexpr std::string_view sourceDirectory{"@IVW_ROOT_DIR@"};
+	inline constexpr std::string_view binaryDirectory{"@IVW_BINARY_DIR@"};
```
Comment (ResearchDaniel) [inline cmake/templates/inviwocommondefines_template.h:8]: Hmm, binaryDirectory is only to be use during development?

File: cmake/installutils.cmake (hunk 2 above)
Comment (ResearchDaniel) [reply cmake/installutils.cmake]: Also, hos does this relate to ivw_default_install_targets

File: cmake/vcpkghelpers.cmake [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+            if(line MATCHES "Homepage: (.*)")
+                set(homepage URL ${CMAKE_MATCH_1})
+            endif()
+        endforeach()
+    else()
+        message(WARNING "Vcpkg control file not found ${portdir}/CONTROL")
+    endif()    
+
+    if(ARG_COPYRIGHT)
+        set(copyright "${pkgdir}/share/${lowercase_name}/${ARG_COPYRIGHT}")
```
Comment (ResearchDaniel) [inline cmake/vcpkghelpers.cmake:151]: Is copyright same as License in vcpkg?

File: ext/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+option(IVW_USE_EXTERNAL_TINYDIR "Tinydir is provided externaly" OFF)
+if(NOT IVW_USE_EXTERNAL_TINYDIR)
+	add_subdirectory(tinydir)
+	ivw_register_license_file(NAME "TinyDir" MODULE Core EXT
+    	URL https://github.com/cxong/tinydir
+    	TYPE "Simplified BSD"
+    	FILES ${IVW_EXTENSIONS_DIR}/tinydir/COPYING
+	)
+else()
+	ivw_vcpkg_paths(INCLUDE vcpkgInclude)
```
Comment (ResearchDaniel) [inline ext/CMakeLists.txt]: It is a bit of a mystery when to use the different vcpkg functions right now

File: ext/libpng/CMakeLists.txt [hunk 10]
```diff
@@ -1,54 +1,38 @@
...
 # Add source files
 file(GLOB_RECURSE SOURCE_FILES ${CMAKE_CURRENT_SOURCE_DIR}/*.c)
 ivw_group("Source Files" ${SOURCE_FILES} )
 
-#--------------------------------------------------------------------
 # Create library
-add_library(libpng ${SOURCE_FILES} ${HEADER_FILES})
-add_library(PNG::PNG ALIAS libpng)
-set_target_properties(libpng PROPERTIES VERSION 1.5.13 SOVERSION 1.5)
+add_library(png ${SOURCE_FILES} ${HEADER_FILES})
```
Comment (ResearchDaniel) [inline ext/libpng/CMakeLists.txt:12]: Why not libpng? Seems to be standard?

File: ext/tinydir/CMakeLists.txt [hunk 11]
```diff
@@ -24,5 +24,5 @@ add_custom_target(tinydircpp SOURCES ${HEADER_FILES})
 # Creates VS folder structure
 ivw_folder(tinydircpp ext)
 
-ivw_default_install_comp_targets(core tinydir)
+#ivw_default_install_targets(tinydir)
```
Comment (ResearchDaniel) [inline ext/tinydir/CMakeLists.txt]: Comment?

File: ext/utf/CMakeLists.txt [hunk 12]
```diff
@@ -25,5 +25,5 @@ set_target_properties(utfcpp PROPERTIES VERSION 3.1.0 SOVERSION 3.1)
 # Creates VS folder structure
 ivw_folder(utfcpp ext)
 
-ivw_default_install_comp_targets(core utf8cpp)
-ivw_make_package(utf8cpp utf8cpp)
\ No newline at end of file
+#ivw_default_install_targets(utf8cpp)
```
Comment (ResearchDaniel) [inline ext/utf/CMakeLists.txt]: Comment?

File: modules/cimg/CMakeLists.txt [hunk 13]
```diff
@@ -31,25 +24,35 @@ set(SOURCE_FILES
...
 find_package(TIFF REQUIRED)
 
+option(IVW_USE_EXTERNAL_CIMG  "Cimg is provided externaly" OFF)
+if(NOT IVW_USE_EXTERNAL_CIMG)
+    add_subdirectory(ext/cimg)
+    ivw_register_license_file(NAME CImg VERSION 2.1.5 MODULE CImg TYPE CeCILL-C
+        URL http://cimg.eu
+        FILES ${CMAKE_CURRENT_SOURCE_DIR}/ext/cimg/Licence_CeCILL-C_V1-en.txt
+    )
+    target_link_libraries(inviwo-module-cimg PRIVATE inviwo::cimg)
```
Comment (ResearchDaniel) [inline modules/cimg/CMakeLists.txt:48]: No vpckg here?

**Review by ResearchDaniel — DISMISSED** [review]
Nice! Sprinkled comments here and there.
Don’t forget to update changelog.
Wishlist: Would be cool if a package could be created for a single module in the future.

File: cmake/installutils.cmake (hunk 1 above)
Comment (petersteneteg) [reply cmake/installutils.cmake:83]: there is one global one directory and one target

File: cmake/installutils.cmake (hunk 2 above)
Comment (petersteneteg) [reply cmake/installutils.cmake]: this is for installing files like python scripts. ivw_default_install_targets is for installing targets i.e. the dlls/exes
Most script should be in the application packages, but if needed we could add an option for component.

File: cmake/installutils.cmake (hunk 3 above)
Comment (petersteneteg) [reply cmake/installutils.cmake:130]: mmm will add... still under development. 
basically it creates a white list for the stiff that you want to install. then in your "app" you can go through that list and remove items/compements etc that you don't want to install

File: cmake/installutils.cmake (hunk 4 above)
Comment (petersteneteg) [reply cmake/installutils.cmake:95]: see above

File: cmake/installutils.cmake (hunk 5 above)
Comment (petersteneteg) [reply cmake/installutils.cmake:201]: think the docs is needed by the app, the help widget...

File: cmake/installutils.cmake (hunk 6 above)
Comment (petersteneteg) [reply cmake/installutils.cmake:164]: but we want the different ones depending on if we are on mac etc?

File: cmake/templates/inviwocommondefines_template.h (hunk 7 above)
Comment (petersteneteg) [reply cmake/templates/inviwocommondefines_template.h:8]: yes, i wanted that to be able to find stuff while developing, when deployed the installer move into the source dir, then it is not needed...

File: cmake/vcpkghelpers.cmake (hunk 8 above)
Comment (petersteneteg) [reply cmake/vcpkghelpers.cmake:151]: yes that is the licence file

File: ext/CMakeLists.txt (hunk 9 above)
Comment (petersteneteg) [reply ext/CMakeLists.txt]: :) will try and and some docs

File: ext/libpng/CMakeLists.txt (hunk 10 above)
Comment (petersteneteg) [reply ext/libpng/CMakeLists.txt:12]: seems png has 2 version, and only one just called png and a new one called libpng. seems though using png still finds the new one...

File: modules/cimg/CMakeLists.txt (hunk 13 above)
Comment (petersteneteg) [reply modules/cimg/CMakeLists.txt:48]: fixed

File: ext/libpng/CMakeLists.txt (hunk 10 above)
Comment (petersteneteg) [reply ext/libpng/CMakeLists.txt:12]: using PNG seems to give the best result at the moment.

File: cmake/templates/inviwocommondefines_template.h (hunk 7 above)
Comment (petersteneteg) [reply cmake/templates/inviwocommondefines_template.h:8]: use this to find the licence files, which I put in the build folder, and then in the source three when we install

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,38 @@
 Here we document changes that affect the public API or changes that needs to be communicated to other developers. 
 
+## 2020-06-26 Vcpkg support
+Most of our external dependencies can now provided by vcpkg. Just install the corresponding vcpkg package and set `IVW_USE_EXTERNAL_<package>` to `TRUE`
```
Comment (ResearchDaniel) [inline CHANGELOG.md]: We should add the vcpkg commands for each dependency and instructions about this to the build instructions. Perhaps vcpkg should be the default way and building it a fallback?

File: cmake/installutils.cmake [hunk 15]
```diff
@@ -45,76 +50,171 @@ if(APPLE)
...
+    list(APPEND install_list 
+        "${CMAKE_CURRENT_BINARY_DIR}|%|${CMAKE_PROJECT_NAME}|%|Application|%|/"
+        "${CMAKE_CURRENT_BINARY_DIR}|%|${CMAKE_PROJECT_NAME}|%|Datasets|%|/"
+        "${CMAKE_CURRENT_BINARY_DIR}|%|${CMAKE_PROJECT_NAME}|%|Testing|%|/"
+        "${CMAKE_CURRENT_BINARY_DIR}|%|${CMAKE_PROJECT_NAME}|%|Development|%|/"
+    )
+    set_property(${scope} PROPERTY INTERFACE_IVW_INSTALL_LIST ${install_list})
+endfunction()
+
+# Filders the given INTERFACE_IVW_INSTALL_LIST for specified components
```
Comment (ResearchDaniel) [inline cmake/installutils.cmake]: Typo

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,5 +1,27 @@
...
+
+## 2020-06-26 CMake refactor
+We have renamed many cmake options to make the naming more consistent and the options easier to find. But you might need to review your cmake settings when updating to make sure you have the correct settings. 
+We now group the cmake settings like this:
+ * `IVW_APP_*` Enable disable building various apps
+ * `IVV_CFG_*` All configuration options, like  PRECOMPILED_HEADERS and PROFILING
+ * `IVW_DOXYGEN_*` Doxygen options
+ * `IVW_EXTERNAL_*` Add external modules / projects
+ * `IVW_MODULE_*` enable/disable modules
+ * `IVW_PACKAGE_*` options for installing/cretings installers
```
Comment (martinfalk) [inline CHANGELOG.md]: typo `creating`

File: ext/CMakeLists.txt [hunk 17]
```diff
@@ -0,0 +1,279 @@
...
+	ivw_register_license_file(NAME libpng TARGET png MODULE Core EXT 
+		TYPE "libpng (BSD-style)" URL http://www.libpng.org/pub/png/libpng.html
+	    FILES ${IVW_EXTENSIONS_DIR}/libpng/LICENSE.txt
+	)
+else()
+	find_package(ZLIB REQUIRED)
+	find_package(JPEG REQUIRED)
+	find_package(TIFF REQUIRED)
+	find_package(PNG  REQUIRED)
+	ivw_vcpkg_install(zlib          MODULE Core EXT)
```
Comment (martinfalk) [inline ext/CMakeLists.txt]: could potentially fail

File: apps/inviwo/CMakeLists.txt [hunk 18]
```diff
@@ -171,14 +146,14 @@ if(WIN32)
...
 ivw_mod_name_to_class(ivw_module_classes ${enabled_modules})
 ivw_private_create_pyconfig("${IVW_MODULE_DIR};${IVW_EXTERNAL_MODULES}" "${ivw_module_classes}" inviwo)
-
-
 ivw_deploy_qt(inviwo)
 
-include(packaging/packaging.cmake) # Package creation
+if(IVW_PACKAGE_SELECT_APP STREQUAL "inviwo")
+    include(packaging/packaging.cmake) # Package creation
+endif()
```
Comment (martinfalk) [inline apps/inviwo/CMakeLists.txt]: missing endline

File: cmake/installutils.cmake [hunk 19]
```diff
@@ -45,76 +50,171 @@ if(APPLE)
...
+set(IVW_PACKAGE_SELECT_APP "inviwo" CACHE STRING "Select which app to package")
+set_property(CACHE IVW_PACKAGE_SELECT_APP PROPERTY STRINGS "inviwo")
+
+# the INTERFACE_IVW_INSTALL_LIST is a property to hold a list of projects/components to install
+# it is used in the packaging to creata a list of items to install. this list will be
+# passed to CPACK_INSTALL_CMAKE_PROJECTS after a potential filtering
+# The property can be set in GLOBAL, DIRECTORY and TARGET scope.
+define_property(
+    GLOBAL PROPERTY INTERFACE_IVW_INSTALL_LIST INHERITED
+    BRIEF_DOCS "List of global intstallation components to install"
```
Comment (martinfalk) [inline cmake/installutils.cmake]: typo "installation" (also next row)

File: cmake/licenses.cmake [hunk 20]
```diff
@@ -29,6 +29,17 @@
 
 include(CMakeParseArguments)
 
+define_property(
+    DIRECTORY PROPERTY IVW_LICENCE_LIST
```
Comment (martinfalk) [inline cmake/licenses.cmake]: British English or American English spelling? Compare also with filename

File: cmake/memleak.cmake [hunk 21]
```diff
@@ -28,15 +28,8 @@
...
-        add_subdirectory(${IVW_EXTENSIONS_DIR}/vld)
-    endif()
-ENDIF()
-
 function(ivw_memleak_setup target)
-    if(WIN32 AND MSVC AND IVW_ENABLE_MSVC_MEMLEAK_TEST)
+    if(WIN32 AND MSVC AND IVW_CFG_MSVC_MEMLEAK_TEST)
         target_link_libraries(${target} PUBLIC vld-interface)
     endif()
 endfunction()
```
Comment (martinfalk) [inline cmake/memleak.cmake]: missing newline

File: cmake/templates/inviwocommondefines_template.h [hunk 22]
```diff
@@ -1,11 +1,10 @@
...
-
-#define IVW_MAJOR_VERSION @IVW_MAJOR_VERSION@
-#define IVW_MINOR_VERSION @IVW_MINOR_VERSION@
-#define IVW_PATCH_VERSION @IVW_PATCH_VERSION@
-#define IVW_BUILD_VERSION @IVW_BUILD_VERSION@
+namespace inviwo::build {
+	inline constexpr std::string_view sourceDirectory{"@IVW_ROOT_DIR@"};
+	inline constexpr std::string_view binaryDirectory{"@IVW_BINARY_DIR@"};	
+	inline constexpr Version version{std::string_view{"@IVW_VERSION@"}};
+}
```
Comment (martinfalk) [inline cmake/templates/inviwocommondefines_template.h]: newline?

File: ext/CMakeLists.txt [hunk 23]
```diff
@@ -0,0 +1,279 @@
...
+	ivw_register_license_file(NAME "Google C++ Testing Framework" TARGET gtest MODULE Core EXT
+        URL "https://github.com/google/googletest"
+        TYPE "BSD 3-clause"
+        FILES ${IVW_EXTENSIONS_DIR}/gtest/gtest/LICENSE
+    )
+else()
+	find_package(GTest CONFIG REQUIRED)
+endif()
+
+# Img foramts
```
Comment (martinfalk) [inline ext/CMakeLists.txt]: typo

File: ext/benchmark/CMakeLists.txt [hunk 24]
```diff
@@ -1,11 +1,9 @@
...
\ No newline at end of file
+# Setup benchmark
+option(BENCHMARK_ENABLE_TESTING "Enable testing of the benchmark library." OFF)
+option(BENCHMARK_ENABLE_INSTALL "Enable installation of benchmark. (Projects embedding benchmark may want to turn this OFF.)" OFF)
+option(BENCHMARK_ENABLE_GTEST_TESTS "Enable building the unit tests which depend on gtest" OFF)
+set(BUILD_SHARED_LIBS OFF)
+add_subdirectory(benchmark)
+set_target_properties(benchmark PROPERTIES FOLDER ext)
+ivw_register_package(benchmark benchmark)
+add_library(benchmark::benchmark ALIAS benchmark)
```
Comment (martinfalk) [inline ext/benchmark/CMakeLists.txt]: newline?

File: ext/flags/CMakeLists.txt [hunk 25]
```diff
@@ -2,22 +2,18 @@
...
+add_library(inviwo::flags ALIAS flags)
 
 #  HACK: have the files showing in the IDE, under the name 'flags'
-add_custom_target(flags SOURCES ${HEADER_FILES})
-
-# Creates VS folder structure
-ivw_folder(flags ext)
\ No newline at end of file
+add_custom_target(flags_dummy SOURCES ${HEADER_FILES})
+ivw_folder(flags_dummy ext)
```
Comment (martinfalk) [inline ext/flags/CMakeLists.txt]: newline?

File: ext/glm/CMakeLists.txt [hunk 26]
```diff
@@ -23,3 +14,4 @@ endif()
 if(TARGET glm_shared)
     ivw_folder(glm_shared ext)
 endif()
+ivw_register_package(glm glm)
```
Comment (martinfalk) [inline ext/glm/CMakeLists.txt]: newline?

File: ext/half/CMakeLists.txt [hunk 27]
```diff
@@ -1,21 +1,13 @@
...
+add_library(half INTERFACE)
+add_library(inviwo::half ALIAS half)
+target_include_directories(half INTERFACE include)
 
-# Creates VS folder structure
-ivw_folder(half ext)
\ No newline at end of file
+#  HACK: have the files showing in the IDE, under the name 'half-dummy'
+add_custom_target(half_dummy SOURCES ${HEADER_FILES})
+ivw_folder(half_dummy ext)
```
Comment (martinfalk) [inline ext/half/CMakeLists.txt]: newline

File: ext/json/CMakeLists.txt [hunk 28]
```diff
@@ -29,4 +29,5 @@ endif()
 #  HACK: have the files showing in the IDE
 add_custom_target(json_dummy SOURCES ${PROJECT_SOURCE_DIR}/single_include/nlohmann/json.hpp)
 # Creates VS folder structure
-ivw_folder(json_dummy ext)
\ No newline at end of file
+ivw_folder(json_dummy ext)
+ivw_register_package(nlohmann_json nlohmann_json)
```
Comment (martinfalk) [inline ext/json/CMakeLists.txt]: newline

File: ext/tclap/CMakeLists.txt [hunk 29]
```diff
@@ -17,5 +17,6 @@ target_include_directories(tclap INTERFACE
 add_custom_target(tclap_dummy SOURCES ${HEADER_FILES})
 set_target_properties(tclap_dummy PROPERTIES VERSION 1.2.1 SOVERSION 1.2)
 
+ivw_register_package(tclap tclap)
 # Creates VS folder structure
 ivw_folder(tclap_dummy ext)
```
Comment (martinfalk) [inline ext/tclap/CMakeLists.txt]: newline

File: ext/ticpp/CMakeLists.txt [hunk 30]
```diff
@@ -46,7 +49,7 @@ ivw_folder(ticpp ext)
 
 #--------------------------------------------------------------------
 # Make package (for other projects to find)
-ivw_default_install_comp_targets(core ticpp)
+ivw_default_install_targets(ticpp)
 ivw_make_package(TiCPP ticpp)
 
 ivw_suppress_compiler_warnings(ticpp)
```
Comment (martinfalk) [inline ext/ticpp/CMakeLists.txt]: newline

File: modules/cimg/CMakeLists.txt [hunk 31]
```diff
@@ -62,57 +66,60 @@ target_compile_definitions(inviwo-module-cimg PRIVATE
...
+option(IVW_USE_OPENEXR  "Add OpenEXR image format support (HDR images)" ON)
+option(IVW_USE_EXTERNAL_OPENEXR  "OpenEXR is provided externaly" OFF)
+if(IVW_USE_OPENEXR)
     target_compile_definitions(inviwo-module-cimg PRIVATE cimg_use_openexr)
+    if(IVW_USE_EXTERNAL_OPENEXR)
+        set(CMAKE_FIND_PACKAGE_PREFER_CONFIG OFF)
+        find_package(OpenEXR REQUIRED)
+        target_link_libraries(inviwo-module-cimg PRIVATE ${OpenEXR_LIBRARIES})
+        set(extraIncDir ${OpenEXR_INCLUDE_DIRS})
+        list(TRANSFORM extraIncDir APPEND "/OpenEXR") # CImg uses includes without OpendEXR
```
Comment (martinfalk) [inline modules/cimg/CMakeLists.txt]: minor typo at end of line `OpenEXR`

File: src/qt/editor/inviwoaboutwindow.cpp [hunk 32]
```diff
@@ -257,46 +258,57 @@ InviwoAboutWindow::InviwoAboutWindow(InviwoMainWindow* mainwindow)
     auto showLicense = [str, textdoc, app, escape, makeBody](const QUrl& url) {
         if (url.hasQuery()) {
             QUrlQuery query(url);
-            auto module = utilqt::fromQString(query.queryItemValue("module"));
-            auto id = utilqt::fromQString(query.queryItemValue("id"));
+            auto moduleName = utilqt::fromQString(query.queryItemValue("module"));
+            auto licenceId = utilqt::fromQString(query.queryItemValue("id"));
```
Comment (martinfalk) [inline src/qt/editor/inviwoaboutwindow.cpp]: spelling

File: src/qt/editor/inviwoaboutwindow.cpp [hunk 33]
```diff
@@ -257,46 +258,57 @@ InviwoAboutWindow::InviwoAboutWindow(InviwoMainWindow* mainwindow)
...
+
+            auto lit = util::find_if(licenses, [&](auto& l) { return l.id == licenceId; });
+            if (lit == licenses.end()) return;
+
+            auto& license = *lit;
+
+            Document doc;
+            auto body = makeBody(doc);
+            auto li = body.append("div");
+            li.append("p").append("a", "Back", {{"href", "file://home.txt"}});
```
Comment (martinfalk) [inline src/qt/editor/inviwoaboutwindow.cpp:278]: home.txt?

**Review by martinfalk — DISMISSED** [review]
nice work!

File: ext/CMakeLists.txt (hunk 17 above)
Comment (petersteneteg) [reply ext/CMakeLists.txt]: should fail in a nice way, i.e. do nothing

File: cmake/licenses.cmake (hunk 20 above)
Comment (petersteneteg) [reply cmake/licenses.cmake]: change to license

File: src/qt/editor/inviwoaboutwindow.cpp (hunk 33 above)
Comment (petersteneteg) [reply src/qt/editor/inviwoaboutwindow.cpp:278]: was there before

File: modules/cimg/CMakeLists.txt [hunk 34]
```diff
@@ -62,57 +66,60 @@ target_compile_definitions(inviwo-module-cimg PRIVATE
...
+            ${CMAKE_CURRENT_BINARY_DIR}/ext/openexr/openexr/IlmBase/config
+            ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/IlmBase/Half
+            ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/IlmBase/Iex
+            ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/IlmBase/IexMath
+            ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/IlmBase/Imath
+            ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/IlmBase/IlmThread
+            ${CMAKE_CURRENT_BINARY_DIR}/ext/openexr/openexr/OpenEXR/config
+            ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/OpenEXR/IlmImf
+            ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/OpenEXR/IlmImfUtil
+        )
```
Comment (ResearchDaniel) [inline modules/cimg/CMakeLists.txt:121]: Vcpkg thingy?

File: CHANGELOG.md (hunk 14 above)
Comment (petersteneteg) [reply CHANGELOG.md]: eventually. I've added vcpkg.json which will become a vcpkg file to do that automatically, but vcpkg does not support it yet.

File: modules/cimg/CMakeLists.txt (hunk 34 above)
Comment (petersteneteg) [reply modules/cimg/CMakeLists.txt:121]: was there before too. OpenEXR does not like to be a submodule...

File: CHANGELOG.md (hunk 14 above)
Comment (petersteneteg) [reply CHANGELOG.md]: added a cmd line to install packages

**Review by martinfalk — DISMISSED** [review]

**Review by martinfalk — APPROVED** [review]

**Pull request merged**